### ISSUE TITLE
Activation variables

### ISSUE THREAD

**Issue by roaldarbol** [prefix-dev/pixi#689]
### Problem description

Currently [activation] in `pixi.toml` contains [scripts]. Currently, if one wishes to set an environmental variable, you need a shell script (e.g. `activate.sh`) tthat contains the new variable - however, it would be great to also have a way of exporting environmental variables directly in `pixi.toml`. Could look like this:
```toml
[activation]
variables = [ 
     "R_LIBS_USER = '$CONDA_PREFIX/lib/R/library'",
     "R_LIBS = \"$CONDA_PREFIX/lib/R/library\"" ,
     "R_LIBS_PROJ = $CONDA_PREFIX/lib/R/library" 
]
```

There's just a few things to iron out I think. 
- Should it `export` by default?
- Should it take the entire expression and run it as a shell command - in which case, how should the ""'s be placed (I've made 3 different ones in the example above)?
- Possibly more

**Comment by roaldarbol** [comment]
I like the `[activation.variables]` approach too! The fewer brackets the better from my point of view. 😅 And I'd say probably expect paths to be relative to the pixi root by default, but with the option of overwriting to an absolute path?

**Comment by roaldarbol** [comment]
Just a stray thought, would something akin to `tasks` be useful too? I.e. a terminal expression? E.g.
```toml
[activation.tasks]
r-activation = "LIB_PATH=\"$CONDA_PREFIX/lib/R/library\"; R_LIBS_PROJ=$LIB_PATH"
```

I don't know if it adds any value over `scripts` - could ask if it would have added value to anyone?

**Comment by ruben-arts** [comment]
Thinking out loud here, we need some logic to see something as a path. As there are multiple types you can save in a variable.
```toml
[activation.variables]
R_LIBS = {path = "path/to/libs"} 
HELLO = {string = "world"}
LIST = {list = ["path/one", "path/two"]} # Possibly the need for appending
```

This feels very verbose but we might need it as we want to be cross shell and platform

**Comment by roaldarbol** [comment]
Hmm, a bit verbose, but I see what you mean. If it's implemented as an alpha feature, we coul dhave it tested out and see if the verbosity gets in the way, and maybe consider alternative ways? Just throwing out the other obvious alternatives:
```toml
[activation.variables]
paths = { R_LIBS = "path/to/lib", R_LIBS_USER = "path/to/lib" }
strings = { HELLO = "world" }
```
or alternatively

```toml
[activation.variables.paths]
// OR
[activation.paths]
R_LIBS = "path/to/lib"
```

**Comment by abkfenris** [comment]
I've found that [Mise's ability to set and modify environment variables](https://mise.jdx.dev/environments.html) has worked pretty well, if you want another Rust tool with a TOML config to compare to.

**Comment by W1M0R** [comment]
@roaldarbol If I understand the issue correctly, this is now implemented (at least for pixi 0.23.0):

```toml
[activation.env]
GOPATH = ""
GOWORK ="off"
GO111MODULE ="on"
```

See https://pixi.sh/latest/reference/project_configuration/#the-activation-table for more information.

**Comment by roaldarbol** [comment]
Ay, that's great, thanks for the pointer! I'll give it a spin tomorrow and close the issue🤙
